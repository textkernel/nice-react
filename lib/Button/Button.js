import React, { Component } from 'react';
import PropTypes from 'prop-types';
import niceProps from '../props';

export default class Button extends Component {
  handleClick(e) {
    e.preventDefault();
    const { onClick } = this.props;
    if (!onClick) {
      return false;
    }
    onClick();
    return true;
  }

  render() {
    const { children, context, size } = this.props;
    const classes = ['nice-btn', context, size].filter(c => c && c.length > 0);

    return (
      <button type="button" onClick={ e => this.handleClick(e) } className={classes.join(' ')}>
        {children}
      </button>
    );
  }
}

Button.propTypes = {
  context: PropTypes.oneOf(niceProps.buttonContexts),
  size: PropTypes.oneOf(niceProps.buttonSizes),
  children: PropTypes.node,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  context: 'primary',
  size: 'medium',
  children: null,
  onClick: null
};
