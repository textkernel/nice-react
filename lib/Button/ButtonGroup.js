import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import props from '../props';

const ButtonGroup = ({ children, className, justified, stacked, style, size, pullRight }) => (
    <div
        className={ classNames('nice-btn-group', {
            className,
            justified,
            stacked,
            [size]: size,
            'pull-right': pullRight
        }) }
        style={ style }
    >
        { children }
    </div>
);

ButtonGroup.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    size: PropTypes.oneOf(props.buttonGroupSizes),
    style: PropTypes.object,
    justified: PropTypes.bool,
    stacked: PropTypes.bool,
    pullRight: PropTypes.bool
};

ButtonGroup.defaultProps = {
    size: null,
    children: null,
    className: '',
    justified: false,
    stacked: false,
    style: null,
    pullRight: false
};

export default ButtonGroup;
