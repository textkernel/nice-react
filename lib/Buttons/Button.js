import React, { Component } from 'react';
import PropTypes from 'prop-types';
import props from '../props';

export default class Button extends Component {
  handleClick(e) {
    e.preventDefault();
    this.props.onClick();
  }

  render() {
    const { children, icon, alternative } = this.props;
    const iconClass = icon ? `icon ${icon}` : null;

    const regularButton = () => {
      const { context, size } = this.props;
      const baseClass = 'nice-btn';
      return [baseClass, context, size, iconClass];
    };

    const alternativeButton = () => {
      const baseClass = 'nice-btn-alt';
      return [baseClass, iconClass];
    };

    const butonClasses = alternative ? alternativeButton : regularButton;
    const classes = butonClasses().filter(c => c && c.length > 0);

    return (
      <a href='#' onClick={e => this.handleClick(e)} className={classes.join(' ')}>
        {children}
      </a>
    );
  }
}

Button.propTypes = {
  context: PropTypes.oneOf(props.buttonContexts),
  size: PropTypes.oneOf(props.buttonSizes),
  children: PropTypes.node,
  onClick: PropTypes.func,
  icon: PropTypes.string,
  alternative: PropTypes.bool
};

Button.defaultProps = {
  context: 'default',
  size: 'medium',
  children: null,
  onClick: null,
  icon: null,
  alternative: false
};