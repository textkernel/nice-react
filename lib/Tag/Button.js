import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';


const Button = ({ icon, className, style, onClick, ...rest }) => {
    const handleClick = (event) => {
        event.stopPropagation();
        onClick(event);
    };

    return (
        <button
            className={classnames('nice-tag-btn', className)}
            style={style}
            onClick={handleClick}
            {...rest}
        >
            <i className={`icon ${icon}`} />
        </button>
    );
};


Button.propTypes = {
    icon: PropTypes.string.isRequired,
    className: PropTypes.string,
    style: PropTypes.object,
    onClick: PropTypes.func,
};


Button.defaultProps = {
    className: '',
    style: {},
    onClick() {},
};


export default Button;