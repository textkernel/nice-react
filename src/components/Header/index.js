export { default } from './Header';
export { default as HeaderMenu } from './Menu';
export { default as HeaderMenuItem } from './Item';
