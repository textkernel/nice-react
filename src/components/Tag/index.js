export { default } from './Tags';
export { default as Tag } from './Tag';