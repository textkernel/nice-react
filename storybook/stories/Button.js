import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { Button } from 'nice-react';

storiesOf('Button', module)
  .add('Default', () => (
    <Button onClick={() => alert('Default')}>Default</Button>
  ))
  .add('Primary', () => (
    <Button context="primary">Primary</Button>
  ))
  .add('Large', () => (
    <Button size="large">Large</Button>
  ))
  .add('Small', () => (
    <Button size="small">Small</Button>
  ));;