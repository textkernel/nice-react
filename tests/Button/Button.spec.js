import React from 'react';
import { Button } from '../../../../../nice-react/src/components/Button';

describe('<Button />', () => {
    it('should render the Button', () => {
        const wrapper = shallow(<Button>Default</Button>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render style Button', () => {
        const wrapper = shallow(<Button context="primary">Primary</Button>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render size Button', () => {
        const wrapper = shallow(<Button size="large">Large</Button>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render style and size Button', () => {
        const wrapper = shallow(<Button context="primary" size="large">Primary large</Button>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should handle the onClick event', () => {
        const onClickHandler = jest.fn();
        const wrapper = shallow(<Button onClick={onClickHandler} context="primary" size="large">Primary large</Button>);
        wrapper.find('button').simulate('click', { preventDefault() {} });

        expect(onClickHandler).toHaveBeenCalled();
    });
});
