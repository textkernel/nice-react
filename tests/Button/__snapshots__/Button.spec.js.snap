// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render size Button 1`] = `
<button
  className="nice-btn default large "
  disabled={false}
  onClick={[Function]}
  style={null}
  type="button"
>
  Large
</button>
`;

exports[`<Button /> should render style Button 1`] = `
<button
  className="nice-btn primary medium "
  disabled={false}
  onClick={[Function]}
  style={null}
  type="button"
>
  Primary
</button>
`;

exports[`<Button /> should render style and size Button 1`] = `
<button
  className="nice-btn primary large "
  disabled={false}
  onClick={[Function]}
  style={null}
  type="button"
>
  Primary large
</button>
`;

exports[`<Button /> should render the Button 1`] = `
<button
  className="nice-btn default medium "
  disabled={false}
  onClick={[Function]}
  style={null}
  type="button"
>
  Default
</button>
`;
